.github/workflows/release.yml

```diff
@@ -0,0 +1,109 @@
+name: kapp-controller release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  kapp-controller-release:
+    name: kapp-controller release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+        with:
+          path: src/github.com/${{ github.repository }}
+
+      - name: Install Carvel Tools
+        uses: vmware-tanzu/carvel-setup-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          only: ytt, kbld
+          ytt: v0.35.1
+          kbld: v0.31.0
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run release script
+        run: |
+          set -e -x
+
+          ./hack/build-release.sh
+
+      - name: Create Release with goreleaser
```

Comment:
why do we need goreleaser in this case?

Reply:
ah i see, seems like for release.yml?

Reply:
might be a simpler alternative here -- somethign along the lines https://github.com/FiloSottile/age/blob/main/.github/workflows/build.yml

Reply:
Creating the release as the trigger mechanism would be useful in removing use of goreleaser, but would also add a manual step of creating a published release. 

I do think goreleaser does do some nice things (e.g. commit history summary) that I am not quite sure how to replicate with that example. But can look into alternative tools for that as well.

Reply:
After discussing further with @cppforlife, it sounds like the best thing to do is move away from goreleaser and support a workflow primarily focused on the following things:

1. Trigger off of publishing a release, rather than off of pushing tag
2. Upload the release.yml to the published release
3. Publish the checksum in the body of the GitHub release
4. Verify the assets (i.e. release.yml) created match via checksum comparison

Reply:
> Trigger off of publishing a release, rather than off of pushing tag

we should still keep the tag trigger imho

Reply:
Updated release process to not use goreleaser. Example of carrying out release here: https://github.com/danielhelfand/carvel-kapp-controller-fork1/releases/tag/v0.31.0

Logs of release here: https://github.com/danielhelfand/carvel-kapp-controller-fork1/runs/4246855321?check_suite_focus=true